tests/testthat/test_matrix_normalization.R

```diff
@@ -0,0 +1,43 @@
+test_that(
+  "Test matrix normalization stan function and its purpose.",
+  {
+    model <- compiled_site_inf_model
+    space_model_fxns <- spatial_fxns$functions
+
+
+    sigma2_generalized <- 0.2
+    dist_matrix <- as.matrix(
+      dist(
+        data.frame(
+          x = c(850, 370, 480, 70),
+          y = c(120, 750, 810, 960),
+          diag = TRUE,
+          upper = TRUE
+        )
+      )
+    )
+    n <- ncol(
+      dist_matrix
+    )
+    corr_matrix <- space_model_fxns$exponential_decay_corr_func(
+      dist_matrix = dist_matrix,
+      phi = 25,
+      l = 1
+    )
+    normalized_corr_matrix <- space_model_fxns$matrix_normalization(
+      corr_matrix
+    )
```

Comment:
Why not check that the normalized matrix has determinant 1

```suggestion
    normalized_corr_matrix <- space_model_fxns$matrix_normalization(
      corr_matrix
    )
    testthat::expect_equal(det(normalized_corr_matrix, 1))
```